py-polars/polars/io/parquet/functions.py

```diff
@@ -145,7 +145,7 @@ def read_parquet(
             )
 
     # Read binary types using `read_parquet`
-    elif isinstance(source, (BinaryIO, BytesIO, bytes)):
+    elif isinstance(source, (io.BufferedIOBase, io.RawIOBase, bytes)):
```

Comment:
This was the culprit - it didn't catch files opened in binary mode.

Reply:
Alright. Nice to have it covered in tests now. :) Would have never seen that in a review. :D

Reply:
Yeah this stuff is mighty confusing with the type hints vs the objects and everything named similarly.

I'm definitely doing more refactoring around the IO functions soon, lots of improvements to be made here.